src/col/vct/col/ast/type/typeclass/TypeImpl.scala

```diff
@@ -44,7 +44,7 @@ trait TypeImpl[G] extends TypeFamilyOps[G] { this: Type[G] =>
       override def succProvider: SuccessorsProvider[G, G] = IdentitySuccessorsProvider
 
       override def dispatch(t: Type[G]): Type[G] = t match {
-        case TVar(Ref(v)) => substitutions(v)
+        case t @ TVar(Ref(v)) => substitutions.get(v).getOrElse(t)
         case other => rewriteDefault(other)
       }
```

Comment:
I understand why you do this, but I think it's a bit bug-prone. Do you think it's possible to add explicit identity mappings where you depend on this behaviour?

Reply:
Turns out I was just not including all type variable assignments up front - once you do, identity mappings should not be necessary if I understand PVL's ad-hoc generics type system correctly.